R/as-data-frame.R

```diff
@@ -192,3 +192,48 @@ as.data.frame.FilterCatalog <- function (x,
                                          ...) {
     catalogToDataFrame(x, keys = keys, row.names = row.names, ...)
 }
+
+#' Head and tail methods for Crunch objects. See [utils::head()] for more details.
+#'
+#' @param x a CrunchDataset, CrunchDataFrame, or CrunchVariable
+#' @param n a single integer representing the length of the returning object.
+#' @param ... ignored
+#' @name head-tail
+#' @aliases head tail
+NULL
+
+#' @rdname head-tail
+#' @export
+setMethod("head", "CrunchDataset", function (x, n=6L, ...) {
+    as.data.frame(x[head(seq_len(nrow(x)), n),], force=TRUE)
+})
+
+#' @rdname head-tail
+#' @export
+setMethod("head", "CrunchDataFrame", function (x, n=6L, ...) {
+    return(head(attr(x, "crunchDataset")))
+})
+
+#' @rdname head-tail
+#' @export
+setMethod("head", "CrunchVariable", function (x, n=6L, ...) {
+    as.vector(x[head(seq_len(length(x)), n)], ...)
+})
+
+#' @rdname head-tail
+#' @export
+setMethod("tail", "CrunchDataset", function (x, n=6L, ...) {
+    as.data.frame(x[tail(seq_len(nrow(x)), n),], force=TRUE)
+})
+
+#' @rdname head-tail
+#' @export
+setMethod("tail", "CrunchDataFrame", function (x, n=6L, ...) {
+    return(tail(attr(x, "crunchDataset")))
+})
```

Comment:
It doesn't look like these would respect the ordering of the CrunchDataFrame which is in the `row.order` attribute.

I don't think it would be too difficult to add that, though we haven't been using that feature too much (it was designed to make merging onto simple feature style geographies possible, but I haven't had the time to push forward on that recently)